notanorm/base.py

```diff
@@ -5,47 +5,74 @@
 
 import time
 import threading
+import logging
 
 from abc import ABC, abstractmethod
 
-import logging
+from .model import DbModel, DbTable
+from . import errors as err
+
 
 log = logging.getLogger(__name__)
-log.setLevel(logging.INFO)
 
 
 def is_list(obj):
-    # be explicit instead of duck typing
-    # otherwise we wind up failing on strings, bytes, and others
-    return (isinstance(obj, list) or
-            isinstance(obj, tuple) or
-            isinstance(obj, set))
+    """Determine if object is list-like, as opposed to string or bytes-like."""
+    return isinstance(obj, (list, set, tuple))
+
+
+def del_all(mapping, to_remove):
+    """Remove list of elements from mapping."""
+    for key in to_remove:
+        del mapping[key]
+
 
+class DbRow(dict):
+    """Default row factory: elements accessible via string or int keys."""
+    __vals = None
 
-class DbRow():
-    def __init__(self, dct={}):
-        for k in dct.keys():
-            setattr(self, k, dct[k])
+    def __init__(self, dct={}):             # pylint: disable=dangerous-default-value
+        self.__dict__ = dct.copy()
+        super().__init__({k.lower(): v for k, v in dct.items()})
 
     def __repr__(self):
-        return "DbRow(" + str(self.__dict__) + ")"
+        return "DbRow(" + super().__repr__() + ")"
 
     def __eq__(self, other):
         return self.__dict__ == other.__dict__
 
+    def __getattr__(self, key):
+        return self[key.lower()]
+
+    def __setattr__(self, key, val):
```

Comment:
Is there a particular reason to allow modification of a row? Also per the other code, seems like it should be `key.lower()`

Reply:
well, we weren't blocking it before, so i people could have been stuffing extra values in db rows